### ISSUE TITLE
Unused parameter compile errors on CentOS 7 (GCC 4.8.)

### ISSUE THREAD

**Issue by danielmewes** [rethinkdb/rethinkdb#5071]
```
In file included from ./src/rdb_protocol/context.hpp:22:0,
                 from src/rdb_protocol/context.cc:2:
./src/rdb_protocol/changefeed.hpp:137:5: warning: unused parameter 'other' [-Wunused-parameter]
     keyspec_t(keyspec_t &&other) = default;
     ^
In file included from ./src/rdb_protocol/query_cache.hpp:22:0,
                 from src/rdb_protocol/context.cc:4:
./src/rdb_protocol/datum_stream.hpp: In constructor 'ql::changespec_t::changespec_t(ql::changefeed::keyspec_t, counted_t<ql::datum_stream_t>)':
./src/rdb_protocol/datum_stream.hpp:66:36: note: synthesized method 'ql::changefeed::keyspec_t::keyspec_t(ql::changefeed::keyspec_t&&)' first required here 
           stream(std::move(_stream)) { }
                                    ^
In file included from ./src/rdb_protocol/context.hpp:22:0,
                 from ./src/rdb_protocol/env.hpp:16,
                 from src/rdb_protocol/profile.cc:14:
./src/rdb_protocol/changefeed.hpp:137:5: warning: unused parameter 'other' [-Wunused-parameter]
     keyspec_t(keyspec_t &&other) = default;
     ^
In file included from ./src/rdb_protocol/env.hpp:17:0,
                 from src/rdb_protocol/profile.cc:14:
./src/rdb_protocol/datum_stream.hpp: In constructor 'ql::changespec_t::changespec_t(ql::changefeed::keyspec_t, counted_t<ql::datum_stream_t>)':
./src/rdb_protocol/datum_stream.hpp:66:36: note: synthesized method 'ql::changefeed::keyspec_t::keyspec_t(ql::changefeed::keyspec_t&&)' first required here 
           stream(std::move(_stream)) { }
                                    ^
In file included from ./src/rdb_protocol/datum_stream.hpp:22:0,
                 from ./src/rdb_protocol/query_cache.hpp:22,
                 from src/rdb_protocol/query_cache.cc:2:
./src/rdb_protocol/changefeed.hpp:137:5: warning: unused parameter 'other' [-Wunused-parameter]
     keyspec_t(keyspec_t &&other) = default;
     ^
In file included from ./src/rdb_protocol/query_cache.hpp:22:0,
                 from src/rdb_protocol/query_cache.cc:2:
./src/rdb_protocol/datum_stream.hpp: In constructor 'ql::changespec_t::changespec_t(ql::changefeed::keyspec_t, counted_t<ql::datum_stream_t>)':
./src/rdb_protocol/datum_stream.hpp:66:36: note: synthesized method 'ql::changefeed::keyspec_t::keyspec_t(ql::changefeed::keyspec_t&&)' first required here 
           stream(std::move(_stream)) { }
```

**Comment by danielmewes** [comment]
Assigning @mlucy.
I think you just need to remove the parameter name for those.

**Comment by danielmewes** [comment]
Disregard that. I'm going to prepare a patch.

**Comment by danielmewes** [comment]
Fixed by e9dd991f111c4a4175e92e3cd7f0171095f34ce9 , though #5073 is still remaining.
